pkg/gitops/client.go

```diff
@@ -0,0 +1,170 @@
+// Copyright 2021 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package gitops
+
+import (
+	"context"
+	"os"
+
+	"antrea.io/antrea/pkg/apis/crd/v1alpha1"
+	networking "k8s.io/api/networking/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type K8sClient struct {
+	client.Client
+}
+
+func NewKubernetes() (*K8sClient, error) {
+	config, err := rest.InClusterConfig()
+	if err != nil {
+		kubeconfig, hasIt := os.LookupEnv("KUBECONFIG")
+		if !hasIt {
+			kubeconfig = clientcmd.RecommendedHomeFile
+		}
+		config, err = clientcmd.BuildConfigFromFlags("", kubeconfig)
+		if err != nil {
+			klog.ErrorS(err, "unable to build config from flags, check that your KUBECONFIG file is correct!")
+			return nil, err
+		}
+	}
+	scheme := runtime.NewScheme()
+	RegisterTypes(scheme)
+	client, err := client.New(config, client.Options{Scheme: scheme})
+	if err != nil {
+		klog.ErrorS(err, "unable to instantiate new generic client")
+		return nil, err
```

Comment:
I'll merge this so we can make progress.

In one of your follow-up PRs, please consider editing this code to adopt this convention: if a function return an error, the expectation is that it will be logged in the caller, and you don't need to also log it in the callee.